openedx/features/termsofservice/static/termsofservice/jsx/Modal.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+import $ from 'jquery';
+import '../css/tos-modal.css'
+
+// Due to an issue with importing modal from paragon, 
+// a modal has been made from scratch instead of importing a new one
+// import { Button, Modal } from '@edx/paragon';
+
+
+class ModalView extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      show: true,
+      tos_html: '',
+      has_user_agreed_to_latest_tos: true,
+      tos_isChecked: false,
+    };
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
+    this.checkboxClicked = this.checkboxClicked.bind(this);
+    this.isCheckboxClicked = this.isCheckboxClicked.bind(this);
+    this.retrievePage();
+  }
+
+  checkboxClicked() {
+    var isChecked = this.state.tos_isChecked;
+    this.setState({ tos_isChecked: !isChecked });
+  }
+
+  isCheckboxClicked() {
+    return !this.state.tos_isChecked;
+  }
+
+  retrievePage() {
+    $.get('/termsofservice/v1/current_tos/')
+      .then(data => {
+        this.setState({
+          tos_html: data.tos_html,
+          has_user_agreed_to_latest_tos: data.has_user_agreed_to_latest_tos
+        });
+        console.log("ayo", this.state.has_user_agreed_to_latest_tos)
+      });
+  }
+
+  showModal() {
+    console.log(this.state);
+    this.setState({ show: true });
+  }
+
+  hideModal() {
+
+    fetch('/termsofservice/v1/current_tos/', {
+      method: 'POST',
+      mode: 'cors',
+      body: JSON.stringify({
+        has_user_agreed: true
+      })
+
+    })
+    this.setState({ show: false });
+  }
+
+  render() {
+
+    if (this.state.has_user_agreed_to_latest_tos) {
+      return (<div></div>)
+    }
+    else {
+      return (
+
+        <main>
+          <Modal show={this.state.show} handleClose={this.hideModal} >
+            <h2 className="mt-3 text-center">Terms of Service Agreement</h2>
+            <p className="text-center">EducateWorkforce has updated its terms of service. Please read the following terms of acknowledge to continue use of the platform.</p>
+
+            <div className="modal-body border border-dark rounded m-3">
+              <div className="scrollable_tos_style"  dangerouslySetInnerHTML={{ __html: (this.state.tos_html) }}></div>
+            </div>
+
+            <div className="modal-footer d-inline text-center">
+              <form>
```

Comment:
@divpatel10 
Please look into adding the `{% csrf_token %}` with the top of the `<form>` element. See more about how this is done here. https://docs.djangoproject.com/en/3.2/ref/csrf/#how-to-use-it

Maybe this is overboard since there really isn't form information being passed to the platform but it would be an additional security measure for CSRF.

Reply:
Maybe we don't need to do what I set above seeing that the check is handled in this class. Please advise.

I think we probably want to add this `csrf_token` should the form ever call an API endpoint like this https://github.com/CUCWD/edx-platform/pull/201/files#diff-21f5be93e9938068f8a5c39358cb195f64d77f94239fc729b0edadfdb1471235R13-R14. This API is done though in this class' constructor and not done through a form by the user where there are input fields.

I think this `@csrf_exempt` was the correct call for your API endpoint that you called in `retrievePage()`.
https://docs.djangoproject.com/en/3.2/ref/csrf/#django.views.decorators.csrf.csrf_exempt

Reply:
I added the csrf_token when a POST request is made to the API

Reply:
@divpatel10 
I'm not sure that you were wrong in having ​@csrf_exempt​ decorator.
[Comparing 67795329991e5e57764e8f88a36990a179259412..924951973b97fe6fa06b43114fbcc031cbb33ffc · CUCWD/edx-platform (github.com)](https://github.com/CUCWD/edx-platform/compare/67795329991e5e57764e8f88a36990a179259412..924951973b97fe6fa06b43114fbcc031cbb33ffc)



Looking at the documentation here [Cross Site Request Forgery protection | Django documentation | Django (djangoproject.com)](https://docs.djangoproject.com/en/3.2/ref/csrf/#django.views.decorators.csrf.csrf_exempt) it appears that if you add that decorator to the view method that it is exempt from protection from the CSRF middleware.

So I guess removing this decorator has the same effect? Not sure honestly.

My thought is that you would add the `{% csrf_token %}` to the form at the top like so, then add the `@requires_csrf_token` decorator to this method https://github.com/CUCWD/edx-platform/pull/201/files#diff-21f5be93e9938068f8a5c39358cb195f64d77f94239fc729b0edadfdb1471235R12
```
<form method="post">{% csrf_token %}
```